lib/Button.js

```diff
@@ -27,6 +29,10 @@ const Button = sdkClass({
     });
     this.panelShowing = false;
   },
+  destroy() {
+    this.button.remove();
+    CustomizableUI.destroyWidget(this.button.id); // remove button from CustomizableUI cache
```

Comment:
@Osmose I think I found a solution. In the destroy method for a button, we use CustomizableUI.destroyWidget to remove the button from the CustomizableUI widget cache. However, we never use CustomizableUI.createWidget, as this doesn't seem to provide any benefit, and only creates side-effects like having the button show up in the "customize" panel. I haven't found any issues or side-effects with this solution yet, and it fixes the problem. Here's where destroyWidget is implemented: https://dxr.mozilla.org/mozilla-central/source/browser/components/customizableui/CustomizableUI.jsm#2427. It looks to me that it removes the button from the cache without any side effects